pkg/fedvolumebackup/restore/restore_manager.go

```diff
@@ -290,15 +297,27 @@ func (rm *restoreManager) waitRestoreTiKVComplete(volumeRestore *v1alpha1.Volume
 		}
 	}
 
+	walCheckCompleted := 0
 	for _, restoreMember := range restoreMembers {
 		restoreMemberName := restoreMember.restore.Name
 		k8sClusterName := restoreMember.k8sClusterName
+		// Here, it is possible restore already complete when warmup strategy is "check-wal-only".
+		if restoreMember.restore.Spec.WarmupStrategy == pingcapv1alpha1.RestoreWarmupStrategyCheckOnly &&
+			pingcapv1alpha1.IsRestoreComplete(restoreMember.restore) {
+			walCheckCompleted += 1
+			continue
+		}
 		if !pingcapv1alpha1.IsRestoreTiKVComplete(restoreMember.restore) {
 			return controller.IgnoreErrorf("restore member %s of cluster %s is not tikv complete", restoreMemberName, k8sClusterName)
 		}
 	}
-	// restore tikv complete
-	if !v1alpha1.IsVolumeRestoreTiKVComplete(volumeRestore) {
+
+	klog.InfoS("all warm up tasks are completed.", "restore", volumeRestore.Name, "wal-check-only-completed", walCheckCompleted, "total", len(restoreMembers))
```

Comment:
If it's async warmup, warmup tasks may not complete here. I suggest you move the logic of check wal complete to a separate function between sync warmup status and check tikv complete.